pkg/uploader/uploader.go

```diff
@@ -50,10 +52,11 @@ func NewUploaderWithClient(log *libbuildpack.Logger, client AMSClient) Uploader
 }
 
 func (up *uploader) Upload(rootDir string, dstURL string) error {
+	up.root = rootDir
```

Comment:
Another option would be to change `Supplier` to take a client/certKeyPair, and call `NewUploader(logger, client, root).Upload()`.
Also not an optimal solution, IMO.

Reply:
I like this idea. It's smoother than the current one

Reply:
It's more hustle than I initially thought. I thinks it's okay like it is